core/accessors.go

```diff
@@ -246,48 +246,40 @@ func DeleteBlockHeaderByNumber(w db.KeyValueWriter, number uint64) error {
 	return w.Delete(db.BlockHeaderByNumberKey(number))
 }
 
-// TODO: Return TransactionReceipt instead of *TransactionReceipt.
-func GetReceiptByHash(
+func GetReceiptByBlockNumberAndIndex(
 	r db.KeyValueReader,
-	hash *felt.TransactionHash,
+	blockNumber uint64,
+	index uint64,
 ) (*TransactionReceipt, error) {
-	val, err := TransactionBlockNumbersAndIndicesByHashBucket.RawValue().Get(r, hash)
+	blockTransactions, err := BlockTransactionsBucket.Get(r, blockNumber)
 	if err != nil {
 		return nil, err
 	}
 
-	receipt, err := ReceiptsByBlockNumberAndIndexBucket.RawKey().Get(r, val)
-	if err != nil {
-		return nil, err
+	if index >= uint64(len(blockTransactions.Receipts)) {
+		return nil, db.ErrKeyNotFound
 	}
-	return &receipt, nil
+
+	return blockTransactions.Receipts[index], nil
 }
 
 func DeleteTxsAndReceipts(batch db.IndexedBatch, blockNum, numTxs uint64) error {
-	// remove txs and receipts
-	for i := range numTxs {
-		key := db.BlockNumIndexKey{
-			Number: blockNum,
-			Index:  i,
-		}
-		txn, err := TransactionsByBlockNumberAndIndexBucket.Get(batch, key)
-		if err != nil {
-			return err
-		}
+	blockTransactions, err := BlockTransactionsBucket.Get(batch, blockNum)
+	if err != nil {
+		return err
+	}
 
-		if err := TransactionsByBlockNumberAndIndexBucket.Delete(batch, key); err != nil {
-			return err
-		}
-		if err := ReceiptsByBlockNumberAndIndexBucket.Delete(batch, key); err != nil {
-			return err
-		}
+	if err := BlockTransactionsBucket.Delete(batch, blockNum); err != nil {
+		return err
+	}
 
-		txHash := (*felt.TransactionHash)(txn.Hash())
+	for _, tx := range blockTransactions.Transactions {
+		txHash := (*felt.TransactionHash)(tx.Hash())
 		if err := TransactionBlockNumbersAndIndicesByHashBucket.Delete(batch, txHash); err != nil {
```

Comment:
nitpick: I know this line is outside the scope of the PR but could you split the assignment from the if